core/trino-main/src/test/java/io/trino/sql/routine/TestSqlRoutineAnalyzer.java

```diff
@@ -124,21 +129,48 @@ void testLanguage()
         assertThat(analyze("FUNCTION test() RETURNS bigint LANGUAGE SQL RETURN abs(-42)"))
                 .returns(true, from(SqlRoutineAnalysis::deterministic));
 
-        assertFails("FUNCTION test() RETURNS bigint LANGUAGE JAVASCRIPT RETURN abs(-42)")
-                .hasErrorCode(NOT_SUPPORTED)
+        assertExtractMetadataFails("FUNCTION test() RETURNS bigint LANGUAGE JAVASCRIPT RETURN abs(-42)")
+                .hasErrorCode(SYNTAX_ERROR)
                 .hasMessage("line 1:52: Only functions using language SQL may be defined using SQL");
 
-        assertFails("FUNCTION test() RETURNS bigint LANGUAGE JAVASCRIPT AS 'xxx'")
-                .hasErrorCode(NOT_SUPPORTED)
-                .hasMessage("line 1:41: Unsupported function language: JAVASCRIPT");
+        assertThatThrownBy(() -> analyze("FUNCTION test() RETURNS bigint LANGUAGE JAVASCRIPT AS 'xxx'"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("function language must be SQL");
 
-        assertFails("FUNCTION test() RETURNS bigint AS 'xxx'")
+        assertExtractMetadataFails("FUNCTION test() RETURNS bigint AS 'xxx'")
                 .hasErrorCode(SYNTAX_ERROR)
                 .hasMessage("line 1:35: Functions using language SQL must be defined using SQL");
 
-        assertFails("FUNCTION test() RETURNS bigint WITH (abc = 'test') RETURN abs(-42)")
+        assertExtractMetadataFails("FUNCTION test() RETURNS bigint WITH (abc = 'test') RETURN abs(-42)")
                 .hasErrorCode(INVALID_FUNCTION_PROPERTY)
                 .hasMessage("line 1:38: Function language SQL does not support properties");
+
+        assertThat(getLanguageName(function("FUNCTION test() RETURNS bigint RETURN 123")))
+                .isEqualTo("SQL");
+
+        assertThat(getLanguageName(function("FUNCTION test() RETURNS bigint LANGUAGE ABC RETURN 123")))
+                .isEqualTo("ABC");
+    }
+
+    @Test
+    void testSecurity()
+    {
+        extractMetadata("FUNCTION test() RETURNS bigint SECURITY INVOKER RETURN 123");
```

Comment:
Do we verify that `SECURITY INVOKER`  is only allowed for SQL functions?

Reply:
We'll add a restriction for now to disallow declaring security for these